### PR TITLE
[WIP] Get project system building on VS15

### PR DESCRIPTION
**Pull request by davkean**
This gets the project system building on VS15 SDK.

Feel free to comment, it's block on the Dev15 images.

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/roslyn-analysis

File: build/Targets/References.targets [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        </Reference>
+      </ItemGroup>
+    </When>
+  </Choose>
+
+  <Choose>
+    <When Condition="'$(IncludeTestReferences)' == 'true'">
+      <ItemGroup>
+
+        <ProjectReference Include="..\Dependencies\Moq.net\Moq.net.csproj">
```
Comment (srivatsn) [inline build/Targets/References.targets]: Should this be $(MSBuildThisFileDirectory)..\..\src\Dependencies?

File: src/Dependencies/VisualStudio/project.json [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
+    
+    "Microsoft.VisualStudio.Shell.Interop.14.0.DesignTime": "14.1.24720",
     "Microsoft.VisualStudio.SDK.EmbedInteropTypes": "14.1.2",
-    "Microsoft.VisualStudio.ManagedInterfaces": "8.0.50727",
-    "Microsoft.VisualStudio.WCFReference.Interop": "9.0.30729",
-    "Microsoft.VisualStudio.Data.Core": "9.0.21022",
-    "Microsoft.VisualStudio.Data.Services": "9.0.21022"
+
+     "Microsoft.VisualStudio.Editor": "15.0.25123-Dev15Preview",
+    "Microsoft.VisualStudio.ImageCatalog": "15.0.25123-Dev15Preview",
```
Comment (srivatsn) [inline src/Dependencies/VisualStudio/project.json]: Indent is off.

File: build/Targets/References.targets (hunk 1 above)
Comment (davkean) [inline build/Targets/References.targets]: :+1:

File: src/Dependencies/VisualStudio/project.json (hunk 2 above)
Comment (davkean) [inline src/Dependencies/VisualStudio/project.json]: :+1:

**Comment by srivatsn** [comment]
👍

**Comment by srivatsn** [comment]
Just saw all the new commits. Will take a look at them later.

**Comment by davkean** [comment]
Hold off, there's going to be ~10 more commits before I'm done.

File: build.cmd [hunk 3]
```diff
@@ -31,7 +33,7 @@ set BinariesDirectory=%Root%binaries\%BuildConfiguration%\
 set LogFile=%BinariesDirectory%Build.log
 if not exist "%BinariesDirectory%" mkdir "%BinariesDirectory%" || goto :BuildFailed
 
-msbuild /nologo /m /consoleloggerparameters:Verbosity=minimal /fileLogger /fileloggerparameters:LogFile="%LogFile%";verbosity=detailed /t:"%MSBuildTarget%" /p:Configuration="%BuildConfiguration%" "%Root%build\build.proj"
+msbuild /nologo /consoleloggerparameters:Verbosity=minimal /fileLogger /fileloggerparameters:LogFile="%LogFile%";verbosity=detailed /t:"%MSBuildTarget%" /p:Configuration="%BuildConfiguration%" "%Root%build\build.proj" %MSBuildAdditionalArguments%
```
Comment (basoundr) [inline build.cmd]: why not `/m`?

File: build/Targets/VSL.Imports.targets [hunk 4]
```diff
@@ -444,5 +445,5 @@
                BinariesDirectory="$(OutDir)"
                IntermediatesDirectory="$(IntermediateOutputPath)"
                Type="$(SignType)" />
-  </Target>
+  </Target>
```
Comment (basoundr) [inline build/Targets/VSL.Imports.targets]: unwanted spaces at the end

File: build.cmd (hunk 3 above)
Comment (davkean) [inline build.cmd]: We set it when not passed the no-multi-proc switch.

File: src/Dependencies/xUnit.net/project.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
   "supports": {},
   "dependencies": {
     "xunit": "2.1.0",
-    "xunit.runner.console": "2.1.0"
+
```
Comment (basoundr) [inline src/Dependencies/xUnit.net/project.json]: blank line

File: src/DeployTestDependencies/project.json [hunk 6]
```diff
@@ -1,6 +1,7 @@
 ﻿{
   "dependencies": {
-    "xunit.runner.visualstudio": "2.1.0"
+    "xunit.runner.visualstudio": "2.1.0",
+	"xunit.runner.console": "2.1.0"
```
Comment (basoundr) [inline src/DeployTestDependencies/project.json]: indent is off

File: build/Targets/VSL.Imports.targets (hunk 4 above)
Comment (davkean) [inline build/Targets/VSL.Imports.targets]: :+1:

File: src/Dependencies/xUnit.net/project.json (hunk 5 above)
Comment (davkean) [inline src/Dependencies/xUnit.net/project.json]: :+1:

File: src/DeployTestDependencies/project.json (hunk 6 above)
Comment (davkean) [inline src/DeployTestDependencies/project.json]: :+1:

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/IProjectThreadingServiceFactory.cs [hunk 7]
```diff
@@ -42,9 +36,14 @@ public JoinableTaskContextNode JoinableTaskContext
                 get { return _context; }
             }
 
+            public JoinableTaskFactory JoinableTaskFactory
```
Comment (basoundr) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/IProjectThreadingServiceFactory.cs]: expression bodies ?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectCapability.cs [hunk 8]
```diff
@@ -1,16 +1,16 @@
...
     /// <summary>
-    ///     Provides common well-known project capabilities.
+    ///     Provides common well-known project flags.
     /// </summary>
     internal static class ProjectCapability
     {
+        public const string AlwaysAvailable = ProjectCapabilities.AlwaysApplicable;
         public const string CSharp = ProjectCapabilities.CSharp;
-        public const string VisualBasic = ProjectCapabilities.VB;
+        public const string VisualBasic = ProjectCapabilities.VB;
```
Comment (basoundr) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectCapability.cs]: unwanted space

**Comment by davkean** [comment]
test ci please

**Comment by davkean** [comment]
test ci please

**Comment by davkean** [comment]
test this please

**Pull request merged**